lib/features/mailbox/data/datasource_impl/mailbox_cache_datasource_impl.dart

```diff
@@ -35,7 +35,9 @@ class MailboxCacheDataSourceImpl extends MailboxDataSource {
   @override
   Future<List<Mailbox>> getAllMailboxCache() {
     return Future.sync(() async {
-      return await _mailboxCacheManager.getAllMailbox();
+      final listMailboxes = await _mailboxCacheManager.getAllMailbox();
+      listMailboxes.sort((mailboxA, mailboxB) => mailboxA.sortOrder!.value.value.compareTo(mailboxB.sortOrder!.value.value));
```

Comment:
There is maybe a better or cleaner way to do this? Like implementing Mailbox or SortOrder with the Comparable interface? (but that would require to modify the jmap-dart-client repo)

Or maybe better ideas? Not sure if this is the best place to do that as well as it means we sort everytime we take the list of mailboxes from the cache, but looked like the easiest.

Well let me know

Reply:
please try with `extensions`, may be you should not modify on dependencies :)

Reply:
Shouldn't the cache store the mailbox sorted on the first place?

Reply:
> please try with extensions, may be you should not modify on dependencies :)

I'm not modifying on dependencies here, that's tmail-flutter code. What do you mean by trying with `extensions`?

> Shouldn't the cache store the mailbox sorted on the first place?

Yeah sounds more logical, but seemed a bit more complicated than just sorting the list when getting it (you have different cases, the getAllMailboxes and /changes). But I could spend more time and try

Reply:
with current db `Hive` maybe we can not store `sorted`. We stored by key and value only

> What do you mean by trying with extensions?

With `extension`, maybe we can implement Mailbox or Sorter with Compare interface without change the code of dependencies.

Reply:
oooooh you mean this? https://dart.dev/guides/language/extension-methods

Nice suggestion, didn't know it existed (I still don't know much about Dart), that sounds like a very cool thing :D I will try^^

Reply:
so the "cache" is an embedded db Hive then?

Reply:
not all, but this cache, we depend on Hive

Reply:
I tried with `extension`, let me know :)